internal/util/function/siliconflow_embedding_provider.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * # Licensed to the LF AI & Data foundation under one
+ * # or more contributor license agreements. See the NOTICE file
+ * # distributed with this work for additional information
+ * # regarding copyright ownership. The ASF licenses this file
+ * # to you under the Apache License, Version 2.0 (the
+ * # "License"); you may not use this file except in compliance
+ * # with the License. You may obtain a copy of the License at
+ * #
+ * #     http://www.apache.org/licenses/LICENSE-2.0
+ * #
+ * # Unless required by applicable law or agreed to in writing, software
+ * # distributed under the License is distributed on an "AS IS" BASIS,
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+package function
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/util/function/models/siliconflow"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+type SiliconflowEmbeddingProvider struct {
+	fieldDim int64
+
+	client        *siliconflow.SiliconflowEmbedding
+	modelName     string
+	embedDimParam int64
+
+	maxBatch   int
+	timeoutSec int64
+}
+
+func createSiliconflowEmbeddingClient(apiKey string, url string) (*siliconflow.SiliconflowEmbedding, error) {
+	if apiKey == "" {
+		apiKey = os.Getenv(siliconflowAKEnvStr)
+	}
+	if apiKey == "" {
+		return nil, fmt.Errorf("Missing credentials. Please pass `api_key`, or configure the %s environment variable in the Milvus service.", siliconflowAKEnvStr)
+	}
+
+	if url == "" {
+		url = "https://api.siliconflow.cn/v1/embeddings"
+	}
+
+	c := siliconflow.NewSiliconflowEmbeddingClient(apiKey, url)
+	return c, nil
+}
+
+func NewSiliconflowEmbeddingProvider(fieldSchema *schemapb.FieldSchema, functionSchema *schemapb.FunctionSchema) (*SiliconflowEmbeddingProvider, error) {
+	fieldDim, err := typeutil.GetDim(fieldSchema)
```

Comment:
looking form the api doc they do not support provide dim in the request, what is the dim of those embedding models of silliconflow? 

if each of their models have a constant dim, should we enforce this earlier? in the current impl if the user sets a incorrect dim, the error is deferred until after some data was embeded.

Reply:
Currently, there is no check at the creation stage to see if field dim and model dim are consistent, only at the runtime stage.

Reply:
Later we can consider adding relevant checks in the creation phase

Reply:
https://github.com/milvus-io/milvus/issues/40004